src/util/extend-xhr.js

```diff
@@ -0,0 +1,31 @@
+import { API_URL } from '/consts'
+
+const xhr = window.XMLHttpRequest
+const nativeOpen = xhr.prototype.open
+
+// You may wish to show a notification, redirect to a new page, or reload all state.
+const logout = () => {
+  window.localStorage.removeItem('token')
+  // window.location.reload()
+}
+
+xhr.prototype.open = function () {
+  this.addEventListener('load', function () {
+    if (this.responseURL.indexOf(API_URL) === -1) return
```

Comment:
If I ran into a situation where this was short-circuiting, I'd greatly appreciate a warning or error in the console so I could figure out why my requests weren't doing anything.

Reply:
Oh haha.. Yeah it should not be returning before calling the nativeOpen 🥀

Reply:
@weslord I'm not sure I understand how this could short-circuit? This is _adding_ an event listener to the load event of the individual xhr instance. It is not replacing the native load handler. So, if the xhr is *not* getting a response from our API_URL then it does nothing extra.

The native xhr `open` handler is still called.

Reply:
Right you are! Whoops.